src/CentrifugeRouter.sol

```diff
@@ -2,20 +2,34 @@
 pragma solidity 0.8.21;
 
 import {Auth} from "src/Auth.sol";
-import {IERC7540} from "src/interfaces/IERC7540.sol";
+import {Multicall} from "src/Multicall.sol";
 import {SafeTransferLib} from "src/libraries/SafeTransferLib.sol";
+import {IERC7540} from "src/interfaces/IERC7540.sol";
+import {IERC20} from "src/interfaces/IERC20.sol";
 import {ICentrifugeRouter} from "src/interfaces/ICentrifugeRouter.sol";
+import {IPoolManager} from "src/interfaces/IPoolManager.sol";
 
-contract CentrifugeRouter is Auth, ICentrifugeRouter {
+contract CentrifugeRouter is Auth, Multicall, ICentrifugeRouter {
     /// @inheritdoc ICentrifugeRouter
     mapping(address user => mapping(address vault => uint256 amount)) public lockedRequests;
+    address public immutable poolManager;
+
+    constructor(address poolManager_) {
+        poolManager = poolManager_;
+    }
 
     // --- Administration ---
     /// @inheritdoc ICentrifugeRouter
```

Comment:
We should add a `file` method for the pool manager. Since the Router is actually somewhat hard to migrate with the locked requests state.